### PR TITLE
Pass TargetFrameworks and PackageTargetFallback to ProjectRestoreInfo

### PR DESCRIPTION
**Pull request by natidea**
Pass TargetFrameworks and PackageTargetFallback to ProjectRestoreInfo to fully enable NuGet restore scenarios. This also refactors useful ConfigurationGeneral Properties into NuGetRestore rule and passes them along as well.

/cc @dotnet/project-system 
/cc @MattGertz for Ask Mode Approval
/cc @alpaix @rrelyea @emgarten for Fyi

### CODE REVIEW

**Comment by MattGertz** [comment]
Approved pending checks & signoff.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs [hunk 1]
```diff
@@ -25,24 +25,24 @@ internal class NuGetRestorer : OnceInitializedOnceDisposedAsync
...
+            .Add(NuGetRestore.SchemaName)
             .Add(ProjectReference.SchemaName)
             .Add(PackageReference.SchemaName)
             .Add(DotNetCliToolReference.SchemaName);
 
         [ImportingConstructor]
         public NuGetRestorer(
             IUnconfiguredProjectVsServices projectVsServices,
-            IVsSolutionRestoreService solutionRestoreService,
+            //IVsSolutionRestoreService solutionRestoreService,
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs]: Why is this commented out?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/NuGetRestore.xaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    <StringProperty Name="MSBuildProjectDirectory" 
+                    Visible="False" 
+                    ReadOnly="True" />
+
+    <StringProperty Name="PackageTargetFallback" 
+                    Visible="False" 
+                    ReadOnly="True" />
+
+    <StringProperty Name="RuntimeIdentifier"
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/NuGetRestore.xaml]: Per Rob's comment we should add RuntimeIdentifiers as well.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs (hunk 1 above)
Comment (natidea) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs]: Fixed, temporary change that should not have been checked in

**Pull request merged**